Deprecate package metadata, improve packaging documentation, refactor record package schema (#1640)
open-contracting/standard: docs/schema/packaging/record_package.md

# Record Package Schema

> The record package schema describes the structure of the container for publishing records. The package contains important metadata.

Reviewer: ```suggestion
The record package schema describes a container for records.
```

````{admonition} Example
:class: hint

> ```{jsoninclude} ../../examples/merging/updates/versioned.json

Reviewer: The two examples are overwhelmingly not package metadata. I would prefer an example that just puts an empty object where other schemas are involved.

The examples were added to the other pages in #1482 to close #1348. They should be kept there, as these specific examples where intended to demonstrate the record and release (in the context of their package).

Author: > The two examples are overwhelmingly not package metadata. I would prefer an example that just puts an empty object where other schemas are involved.

There's no `expand` option specified so the rendered result looks the same and has the advantage of actually being valid data (unlike an empty record object):

![image](https://github.com/open-contracting/standard/assets/19265814/2b5bcbc3-ba06-41d0-8b76-8832f406606b)

Happy to replace it with a minimal example with an empty object, if you still prefer that option.

> The examples were added to the other pages in #1482 to close #1348. They should be kept there, as these specific examples where intended to demonstrate the record and release (in the context of their package).

The sections in which the examples appeared no longer exist. The examples were introduced with a sentence like "Records must be published within a [record package](https://standard.open-contracting.org/staging/1.2-dev/en/schema/record_package/)." Now that packages are optional, I figured it makes sense to provide examples demonstrating the relationship between releases/records and packages on the package reference pages. Where would you want to put the package examples on the release and record reference pages and what framing should we provide?

If we choose to leave them off the release and record pages, I think we should add an example of a record to the record reference page in order to demonstrate what a full record looks like. (The release structure section of the release reference already includes an example of a release.)

Reviewer: > There's no expand option specified so the rendered result looks the same

Ah, perfect. That works.

> If we choose to leave them off the release and record pages, I think we should add an example of a record to the record reference page in order to demonstrate what a full record looks like.

Yes, sounds good.

> The sections in which the examples appeared no longer exist.

Oh, right. Your changes sound fine then.

Just a note to myself to take a look at the rendered pages when doing a full review.

Author: > > If we choose to leave them off the release and record pages, I think we should add an example of a record to the record reference page in order to demonstrate what a full record looks like.
> 
> Yes, sounds good.

Done in https://github.com/open-contracting/standard/pull/1640/commits/ddfdc0c12406222f14870afe0ebba865b8db02d8

:jsonpointer:
:title: Record package
```
````

> For this version of OCDS, the canonical URL for the record package schema is <https://standard.open-contracting.org/schema/1__1__5/record-package-schema.json> and for the versioned release schema is <https://standard.open-contracting.org/schema/1__1__5/versioned-release-validation-schema.json>. Using the canonical URL guarantees that your software, documentation or other resources will always refer to the specific version of the schema with which they were authored and tested.

Reviewer: Versioned release schema is moved to record schema.

```suggestion
For this version of OCDS, the canonical URL for the record package schema is <https://standard.open-contracting.org/schema/1__1__5/record-package-schema.json>. Using the canonical URL guarantees that your software, documentation or other resources will always refer to the specific version of the schema with which they were authored and tested.
```

> In addition to publishing the structured metadata described by the record package schema, you ought to license your data and provide guidance for data users. For more information, see the [licensing guidance](../../guidance/publish.md#license-your-data) and [publication policy guidance](../../guidance/publish.md#finalize-your-publication-policy).

Reviewer: ```suggestion
In addition to publishing the metadata described by the record package schema, you ought to license your data and guide data users. For more information, see the [licensing](../../guidance/publish.md#license-your-data) and [publication policy](../../guidance/publish.md#finalize-your-publication-policy) guidance.
```

```{admonition} Browsing the schema
:class: note

> This page presents the record package schema in an interactive browser and in a table, with additional information in paragraphs. You can also download the canonical version of the record package schema as [JSON Schema](../../../build/current_lang/record-package-schema.json).

Reviewer: We don't have additional paragraphs here.

```suggestion
This page presents the record package schema in an interactive browser and in a table. You can also download the canonical version of the record package schema as [JSON Schema](../../../build/current_lang/record-package-schema.json).
```
